benchmark/Snakefile

```diff
@@ -0,0 +1,537 @@
+
+rule dadi_generate_cache:
+    input:
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.InferDM.bestfits"
+    output:
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.spectra.bpkl"
+    threads: 8
+    resources: time_min=60, mem_mb=5000, cpus=8
+    params:
+        sample_size = 100
+    shell:
+        "dadi-cli GenerateCache --model two_epoch --demo-popt {input} --sample-size {params.sample_size} --output {output} --mp --grids 800 1000 1200 --gamma-pts 2000"
+
+rule dadi_infer_dfe:
+    input:
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.nonneu.fs",
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.spectra.bpkl",
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.InferDM.bestfits"
+    output:
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.gamma.InferDFE.opts.0"
+    threads: 8
+    resources: time_min=60, mem_mb=5000, cpus=8
+    shell:
+        "dadi-cli InferDFE --fs {input[0]} --cache1d {input[1]} --demo-popt {input[2]} --output-prefix {output_dir}/dadi/{wildcards.seeds}/{wildcards.chrms}/pop{wildcards.ids}/pop{wildcards.ids}.two_epoch.gamma --pdf1d {dfe_model} --p0 {dfe_p0} --ubounds {dfe_ubounds} --lbounds {dfe_lbounds} --ratio 2.31 --optimizations 500"
+
+rule dadi_bestfit_dfe:
+    input:
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.gamma.InferDFE.opts.0"
+    output:
+        output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.two_epoch.gamma.InferDFE.bestfits"
+    threads: 1
+    resources: time_min=60, mem_mb=5000, cpus=1
+    shell:
+        "dadi-cli BestFit --input-prefix {output_dir}/dadi/{wildcards.seeds}/{wildcards.chrms}/pop{wildcards.ids}/pop{wildcards.ids}.two_epoch.gamma.InferDFE --pdf gamma --ubounds {dfe_ubounds} --lbounds {dfe_lbounds}"
+
+rule combine_dadi_dfe_bestfits:
+    input:
+        pop0_res = expand(output_dir + "/dadi/{seeds}/{chrms}/pop0/pop0.two_epoch.gamma.InferDFE.bestfits", seeds=seed_array, chrms=chrm_list),
+        pop1_res = expand(output_dir + "/dadi/{seeds}/{chrms}/pop1/pop1.two_epoch.gamma.InferDFE.bestfits", seeds=seed_array, chrms=chrm_list),
+        pop2_res = expand(output_dir + "/dadi/{seeds}/{chrms}/pop2/pop2.two_epoch.gamma.InferDFE.bestfits", seeds=seed_array, chrms=chrm_list)
+    output:
+        expand(output_dir + "/dadi/{chrms}/pop{ids}.dadi.dfe.bestfits", chrms=chrm_list, ids=[0, 1, 2])
+    threads: 1
+    shell:
+        """
+        for l in {input.pop0_res}; do 
+            grep -v '#' $l | head -1
+        done | sed '1i#likelihood\tshape\tscale\ttheta' > {output[0]}
+        for l in {input.pop1_res}; do 
+            grep -v '#' $l | head -1
+        done | sed '1i#likelihood\tshape\tscale\ttheta' > {output[1]}
+        for l in {input.pop2_res}; do 
+            grep -v '#' $l | head -1
+        done | sed '1i#likelihood\tshape\tscale\ttheta' > {output[2]}
+        """
```

Comment:
Hi @chriscrsmith, here are the codes for DFE inference with dadi.
I used dadi through the command line inferface dadi-cli
You can find how to use dadi-cli from the README in https://github.com/xin-huang/dadi-cli
Please install the latest version in the github repo. Thank you.

Reply:
The config file for slurm can be found in https://github.com/xin-huang/analysis2/blob/main/benchmark/config/snakemake/slurm/config.yaml
To run snakemake in slurm, use `snakemake --profile ./config/snakemake/slurm/`